0xC0FFEEEE - O365 Suspicious Mailbox Rule Created

0xC0FFEEEE opened the pull request:
### Details

Adding a detection for a common technique employed during BEC.

### Checklist

- [ ] Validate name matches `<platform>_<mitre att&ck technique>_<short description>` nomenclature
- [ ] [CI/CD](https://github.com/splunk/security_content/actions) jobs passed ✔️ 
- [ ] Validated SPL logic.
- [ ] Validated tags, description, and how to implement.
- [ ] Verified references match analytic.
- [ ] Confirm updates to lookups are handled properly.

### Notes For Submitters and Reviewers

Associated test data - splunk/attack_data#962

0xC0FFEEEE committed:
O365 Suspicious Mailbox Rule Created

patel-bhavin committed:
Merge branch 'develop' into susmailrule

patel-bhavin commented:
Hello @0xC0FFEEEE : Thank you for the PR!  We have a similar detection in the repo - O365 New Email Forwarding Rule Created . Curious to know if we can improve the current one here : 

https://github.com/splunk/security_content/blob/a33941597ee0a67888e73ee90eec1a5845b92b5a/detections/cloud/o365_new_email_forwarding_rule_created.yml#L9

0xC0FFEEEE commented:
Hey @patel-bhavin 

I think the intent of this detection is different enough that both can co-exist.

I created this one with the specific aim of alerting on the defense evasion technique T1564.008, whereas the existing detection covers T1114.003.

It's also worth pointing out that the brilliant and prolific @nterl0k has an open PR #3292 with a very similar detection as well. I've tested both side by side against our production data and mine produced fewer false positives over a 7 day period by quite a margin, something like 350 vs 5.

Happy to take your lead, but I think it might be challenging to merge both, or all 3 detections without muddying the waters between two very different attacker intents.

I think there's potentially a place for all 3, with @nterl0k's as a hunting rule, thoughts?

nterl0k commented:
Ironically the situation that 0xC0FFEEEE's rule detects is the exact situation I developed my rule for... But I didn't want to restrict it to a very specific actor TTP.

I agree with 0xC0FFEEEE, I'm happy to tag mine as a hunting rule since it's pretty wide in its coverage.


-------- Original message --------
From: 0xC0FFEEEE ***@***.***>
Date: 2/18/25 6:53 PM (GMT-05:00)
To: splunk/security_content ***@***.***>
Cc: Steven Dick ***@***.***>, Mention ***@***.***>
Subject: Re: [splunk/security_content] 0xC0FFEEEE - O365 Suspicious Mailbox Rule Created (PR #3336)


Hey @patel-bhavin<https://github.com/patel-bhavin>

I think the intent of this detection is different enough that both can co-exist.

I created this one with the specific aim of alerting on the defense evasion technique T1564.008, whereas the existing detection covers T1114.003.

It's also worth pointing out that the brilliant and prolific @nterl0k<https://github.com/nterl0k> has an open PR #3292<https://github.com/splunk/security_content/pull/3292> with a very similar detection as well. I've tested both side by side against our production data and mine produced fewer false positives over a 7 day period by quite a margin, something like 350 vs 5.

Happy to take your lead, but I think it might be challenging to merge both, or all 3 detections without muddying the waters between two very different attacker intents.

I think there's potentially a place for all 3, with @nterl0k<https://github.com/nterl0k>'s as a hunting rule, thoughts?

—
Reply to this email directly, view it on GitHub<https://github.com/splunk/security_content/pull/3336#issuecomment-2667182621>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJIYP7X3JUQJCRYRZHIMML32QPBXTAVCNFSM6AAAAABXFLN2ICVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNRXGE4DENRSGE>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

[0xC0FFEEEE]0xC0FFEEEE left a comment (splunk/security_content#3336)<https://github.com/splunk/security_content/pull/3336#issuecomment-2667182621>

Hey @patel-bhavin<https://github.com/patel-bhavin>

I think the intent of this detection is different enough that both can co-exist.

I created this one with the specific aim of alerting on the defense evasion technique T1564.008, whereas the existing detection covers T1114.003.

It's also worth pointing out that the brilliant and prolific @nterl0k<https://github.com/nterl0k> has an open PR #3292<https://github.com/splunk/security_content/pull/3292> with a very similar detection as well. I've tested both side by side against our production data and mine produced fewer false positives over a 7 day period by quite a margin, something like 350 vs 5.

Happy to take your lead, but I think it might be challenging to merge both, or all 3 detections without muddying the waters between two very different attacker intents.

I think there's potentially a place for all 3, with @nterl0k<https://github.com/nterl0k>'s as a hunting rule, thoughts?

—
Reply to this email directly, view it on GitHub<https://github.com/splunk/security_content/pull/3336#issuecomment-2667182621>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJIYP7X3JUQJCRYRZHIMML32QPBXTAVCNFSM6AAAAABXFLN2ICVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNRXGE4DENRSGE>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

pyth0n1c commented on detections/cloud/o365_suspicious_mailbox_rule_created.yml:
```diff
@@ -0,0 +1,57 @@
+author: 0xC0FFEEEE
+type: TTP
+status: production
+enabled_by_default: true
```
This **MUST** be removed from the YML. We do not ship any detections enabled by default.
I have switched this to DRAFT to prevent accidental merge.

While this is technically a valid key, at this time this is NEVER something we want to do in ESCU (but is a legitimate use case for folks who may be using contentctl for other repos or internal apps).
Until we make an intentional decision to change this, we should consider including a check in our CICD which throws errors is a rule is enabled_by_default in this repo.

0xC0FFEEEE replied:
My mistake, I copied this YAML over from our DAC repo and completely missed this.

pyth0n1c replied:
@0xC0FFEEEE I want to apologize as, in rereading this,  my comment comes off as harsh.  You are a frequent contributor to the project (and contentctl) and we appreciate your contributions.

I meant to communicate to my team during the review process that we currently lack mechanism to enforce this restriction in this repo (and should consider such checks in the future), but should have made this point in a more clear and constructive way.

0xC0FFEEEE replied:
@pyth0n1c no offence taken!

I'm annoyed I didn't spot this myself, we've actually just taken the same decision not to enable searches by default so I should have thought to double check.

I imagine DaC adoption w/ contentctl is steadily increasing, so just as well you picked up on this and identified the need to check this in your CICD pipeline as it likely won't be the last time this happens, though hopefully not from myself next time!

pyth0n1c reviewed (changes requested):
We do not ship ANY detections enabled_by_Default.

0xC0FFEEEE committed:
remove enabled_by_default

0xC0FFEEEE committed:
Merge branch 'develop' into susmailrule

patel-bhavin committed:
minor changes to ensure CI testing works

patel-bhavin committed:
Merge branch 'develop' into susmailrule

patel-bhavin commented:
@nterl0k @0xC0FFEEEE - thanks both for quickly jumping on deconflicting these detections and details. I largely agree with you both and i think @nterl0k  we should make the other detection as hunting and perhaps keep all three detections.  Changing this to hunting will also help simplify the purpose of different types of same analytic!

pyth0n1c committed:
Merge branch 'develop' into susmailrule

pyth0n1c commented on detections/cloud/o365_suspicious_mailbox_rule_created.yml:
```diff
@@ -0,0 +1,57 @@
+  "^(RSS|Conversation History|Archive)"), 1, 0) | eval suspicious_score=entropy_score+len_score+read_score+folder_score
+  | where suspicious_score>2 | `o365_suspicious_mailbox_rule_created_filter`'
+how_to_implement: You must install the Splunk Microsoft Office 365 Add-on and ingest
+  Office 365 management activity events.
```
We should probably update how_to_implement to indicate the the URL Toolbox App https://splunkbase.splunk.com/app/2734  is required for the `ut_shannon_lookup` to work.
We seem to do this in other content that requires it:
https://github.com/splunk/security_content/blob/32c10dea2638905dcf499f5e43b8a85a98b493a1/detections/endpoint/potential_password_in_username.yml#L34
https://github.com/splunk/security_content/blob/32c10dea2638905dcf499f5e43b8a85a98b493a1/detections/endpoint/randomly_generated_scheduled_task_name.yml#L23
https://github.com/splunk/security_content/blob/32c10dea2638905dcf499f5e43b8a85a98b493a1/detections/endpoint/randomly_generated_windows_service_name.yml#L22

pyth0n1c reviewed (dismissed):
Just some small changes to the how to implement field.

0xC0FFEEEE committed:
add url toolbox requirement

josehelps commented on detections/cloud/o365_suspicious_mailbox_rule_created.yml:
```diff
@@ -0,0 +1,58 @@
+name: O365 Suspicious Mailbox Rule Created
```
I would love to avoid the terminology/classification of "suspicious" or "anomalous" since it is hard to understand what exactly makes it so, can we change the title to "o365 Compromised Mailbox Rule Created" or something that includes "BEC" in the title/description. Otherwise thank you so much @0xC0FFEEEE this is an awesome rule to add!

0xC0FFEEEE replied:
I feel that it's hard to avoid the usage of "suspicious" or "anomalous", as particularly the former feels very natural to say when referring to a detection that isn't neccesarily a conclusive indicator that something bad has occurred.

How about `O365 Potential BEC Email Hiding Rule`, or `O365 BEC Email Hiding Rule Created`? These lean in to the technique name and is nice and succinct. Or alternatively, ChatGPT suggested `O365 BEC Mailbox Rule Created for Evasion`? I like that one less tbh, possibly because it speaks to the motive/tactic itsself as opposed to the technique employed.

josehelps replied:
O365 BEC Email Hiding Rule Created is a solid choice. Thank you for considering the change!

0xC0FFEEEE committed:
Update detection name

0xC0FFEEEE committed:
Merge branch 'develop' into susmailrule

patel-bhavin committed:
Merge branch 'develop' into susmailrule

dluxtron commented:
I have no idea why I'm here right looking at this right now, but would it be possible to include the rule Name as the desc threat object like @nterl0k does?

Very good linkage, the past few campaigns I saw all reused rule names. I could see the rule names being used in threat intel. 

I like these detections a lot.

0xC0FFEEEE committed:
simplify SPL & add threat object

0xC0FFEEEE commented:
@dluxtron thanks for your feedback!

Added the rule name as a threat object as requested.

I also realized I didn't need to jump through various hoops to extract the nested `Parameters.Names` & `Parameters.Values` fieldnames/values using mvzip & mvexpand so I've simplified the SPL while I was there.

0xC0FFEEEE committed:
Merge branch 'develop' into susmailrule

patel-bhavin committed:
Merge branch 'develop' into susmailrule

patel-bhavin committed:
Merge branch 'develop' into susmailrule

patel-bhavin commented:
@0xC0FFEEEE - thank you for incorporating all the various feedback on this PR! We should be good to ship this in ESCU 5.2.0!

patel-bhavin committed:
drilldown update- minor
